src/OpenFeature.Contrib.Providers.EnvVar/EnvVarProvider.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenFeature.Constant;
+using OpenFeature.Error;
+using OpenFeature.Model;
+
+namespace OpenFeature.Contrib.Providers.EnvVar
+{
+    /// <summary>
+    /// An OpenFeature provider using environment variables. 
+    /// </summary>
+    public class EnvVarProvider : FeatureProvider
+    {
+        private const string Name = "Environment Variable Provider";
+        private readonly string _prefix;
+        private delegate bool TryConvert<TResult>(string value, out TResult result); 
+
+        /// <summary>
+        /// Creates a new instance of <see cref="EnvVarProvider"/>  
+        /// </summary>
+        public EnvVarProvider() : this("")
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="EnvVarProvider"/>  
+        /// </summary>
+        /// <param name="prefix">A prefix which will be used when evaluating environment variables</param>
+        public EnvVarProvider(string prefix)
+        {
+            _prefix = prefix; 
+        }
+    
+        /// <inheritdoc/>
+        public override Metadata GetMetadata()
+        {
+            return new Metadata(Name);
+        }
+    
+        private Task<ResolutionDetails<T>> Resolve<T>(string flagKey, T defaultValue, TryConvert<T> tryConvert)
+        {
+            var envVarName = $"{_prefix}{flagKey}";  
+            var value = Environment.GetEnvironmentVariable(envVarName);
+
+            if (value == null)
+                return Task.FromResult(new ResolutionDetails<T>(flagKey, defaultValue, ErrorType.None, Reason.Default));
```

Comment:
I don't think this is the desired approach. In this case, you return the `DefaultValue` without letting the consumer know it is a default value. I would prefer this to return as `ErrorType.FlagNotFound`.

Reply:
Yes! Please return an error in the `ResolutionDetails`. We're trying to update all providers to do that because it's more performant and helps us support flag metadata in error conditions.